backend/services/inventory/api/http/api_inventory.go

```diff
@@ -403,155 +271,193 @@ func (i *inventoryHandlers) GetDevicesHandler(w rest.ResponseWriter, r *rest.Req
+			errors.New("internal error"),
+		)
 		return
 	}
 
-	w.WriteHeader(http.StatusNoContent)
+	c.Status(http.StatusNoContent)
 }
 
-func (i *inventoryHandlers) DeleteDeviceHandler(w rest.ResponseWriter, r *rest.Request) {
-	ctx := r.Context()
-	tenantId := r.PathParam("tenant_id")
+func (i *InternalAPI) DeleteDeviceHandler(c *gin.Context) {
+	ctx := c.Request.Context()
+	tenantId := c.Param("tenant_id")
 	if tenantId != "" {
-		id := &midentity.Identity{
+		id := &identity.Identity{
 			Tenant: tenantId,
 		}
-		ctx = midentity.WithContext(ctx, id)
+		ctx = identity.WithContext(ctx, id)
 	}
 
-	l := log.FromContext(ctx)
+	deviceID := c.Param("device_id")
 
-	deviceID := r.PathParam("device_id")
-
-	err := i.inventory.DeleteDevice(ctx, model.DeviceID(deviceID))
+	err := i.App.DeleteDevice(ctx, model.DeviceID(deviceID))
 	if err != nil && err != store.ErrDevNotFound {
-		u.RestErrWithLogInternal(w, r, l, err)
+		rest.RenderError(c,
+			http.StatusInternalServerError,
+			errors.New("internal error"),
+		)
 		return
 	}
 
-	w.WriteHeader(http.StatusNoContent)
+	c.Status(http.StatusNoContent)
 }
 
-func (i *inventoryHandlers) AddDeviceHandler(w rest.ResponseWriter, r *rest.Request) {
-	ctx := r.Context()
-	tenantId := r.PathParam("tenant_id")
+func (i *InternalAPI) AddDeviceHandler(c *gin.Context) {
+	ctx := c.Request.Context()
+	tenantId := c.Param("tenant_id")
 	if tenantId != "" {
-		id := &midentity.Identity{
+		id := &identity.Identity{
 			Tenant: tenantId,
 		}
-		ctx = midentity.WithContext(ctx, id)
+		ctx = identity.WithContext(ctx, id)
 	}
 
-	l := log.FromContext(ctx)
-
-	dev, err := parseDevice(r)
+	dev, err := parseDevice(c)
 	if err != nil {
-		u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+		rest.RenderError(c,
+			http.StatusBadRequest,
+			err,
+		)
 		return
 	}
 
 	err = dev.Attributes.Validate()
 	if err != nil {
-		u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+		rest.RenderError(c,
+			http.StatusBadRequest,
+			err,
+		)
 		return
 	}
 
-	err = i.inventory.AddDevice(ctx, dev)
+	err = i.App.AddDevice(ctx, dev)
 	if err != nil {
-		u.RestErrWithLogInternal(w, r, l, err)
+		rest.RenderError(c,
+			http.StatusInternalServerError,
+			errors.New("internal error"),
+		)
 		return
 	}
 
-	w.Header().Add("Location", "devices/"+dev.ID.String())
-	w.WriteHeader(http.StatusCreated)
+	c.Writer.Header().Add("Location", "devices/"+dev.ID.String())
+	c.Status(http.StatusCreated)
 }
 
-func (i *inventoryHandlers) UpdateDeviceAttributesHandler(w rest.ResponseWriter, r *rest.Request) {
-	ctx := r.Context()
-	l := log.FromContext(ctx)
+func (i *ManagementAPI) UpdateDeviceAttributesHandler(c *gin.Context) {
+	ctx := c.Request.Context()
+
 	var idata *identity.Identity
 	if idata = identity.FromContext(ctx); idata == nil || !idata.IsDevice {
-		u.RestErrWithLog(w, r, l, errors.New("unauthorized"), http.StatusUnauthorized)
+		rest.RenderError(c,
+			http.StatusUnauthorized,
+			errors.New("unauthorized"),
+		)
```

Comment:
Same here, you could do:
```suggestion
		rest.RenderErrorWithMessage(c,
			http.StatusUnauthorized,
			err, "unauthorized",
		)
```

Reply:
There is no error to return here. `err` is not defined.

Reply:
Sorry, you're right. Please disregard this one.